Makefile

```diff
@@ -153,60 +133,85 @@ define fpm-debian-build =
 	fpm -a $$PKG_ARCH -s dir -t deb -n lantern-manoto -v $$VERSION -m "$(PACKAGE_MAINTAINER)" --description "$(LANTERN_DESCRIPTION)\n$(LANTERN_EXTENDED_DESCRIPTION)" --category net --license "Apache-2.0" --vendor "$(PACKAGE_VENDOR)" --url $(PACKAGE_URL) --deb-compression xz -f -C $$WORKDIR usr;
 endef
 
+define docker-up
+	if [[ "$$(uname -s)" == "Darwin" ]]; then \
+		if [[ -z "$(DOCKERMACHINE)" ]]; then \
+		  if [[ -z "$(BOOT2DOCKER)" ]]; then \
+  			echo 'Missing "docker-machine" command' && exit 1; \
+			fi && \
+			echo "Falling back to using $(BOOT2DOCKER), recommend upgrading to latest docker toolbox from https://www.docker.com/docker-toolbox" && \
+			if [[ "$$($(BOOT2DOCKER) status)" != "running" ]]; then \
+				$(BOOT2DOCKER) up; \
+			fi && \
+			if [[ -z "$$DOCKER_HOST" ]]; then \
+				$$($(BOOT2DOCKER) shellinit 2>/dev/null); \
+			fi \
+		else \
+		  echo "Using $(DOCKERMACHINE)" && \
+			if [[ "$$($(DOCKERMACHINE) status default)" != "Running" ]]; then \
+				$(DOCKERMACHINE) start default; \
+			fi && \
+			$$($(DOCKERMACHINE) env default 2>/dev/null | head -4 | tr -d '"'); \
+		fi \
+	fi
+endef
+
+# This implicit rule allows prefix any existing target with "docker-" to make it
+# run in docker.
+docker-%: system-checks
+	@$(call docker-up) && \
+	DOCKER_CONTEXT=.$(DOCKER_IMAGE_TAG)-context && \
+	mkdir -p $$DOCKER_CONTEXT && \
+	cp Dockerfile $$DOCKER_CONTEXT && \
+	docker build -t $(DOCKER_IMAGE_TAG) $$DOCKER_CONTEXT && \
+	docker run -v $$PWD:/lantern -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /lantern && VERSION="'$$VERSION'" HEADLESS="'$$HEADLESS'" SECRETS_DIR="'$$SECRETS_DIR'" BNS_CERT_PASS="'$$BNS_CERT_PASS'" make $*';
+
 all: binaries
 android-dist: genconfig android
 
-# This is to be called within the docker image.
-docker-genassets: require-npm
+$(RESOURCES_DOT_GO): $(NPM)
```

Comment:
Hehe, I never actually included that change because you and @xiam convinced me that it was best to handle makefile changes separately.  The important thing is that the packages target runs clean-desktop, which cleans this and forces the assets to be regenerated, so for official builds we're good either way.